back/app/services/admin_publications_filtering_service.rb

```diff
@@ -41,11 +41,25 @@ def for_homepage_filter(scope)
     projects = Project.where(id: scope.where(publication_type: Project.name).select(:publication_id))
     filtered_projects = ProjectsFilteringService.new.filter(projects, options)
 
+    if options[:search].present?
+      filtered_projects = filtered_projects.search_by_all(options[:search])
+    end
+
     project_publications = scope.where(publication: filtered_projects)
     other_publications = scope.where.not(publication_type: Project.name)
     project_publications.or(other_publications)
   end
 
+  add_filter('filter_folders') do |scope, options|
+    next scope if options[:search].blank?
+
+    searched_folders = ProjectFolders::Folder.search_by_all(options[:search])
```

Comment:
This line is throwing errors in your specs. I think it's because your are calling `search_by_all` on a class name and not on a collection of objects, as in the project example on line 45.

I haven't really thought this through, but maybe something like:
```
folders = scope.where(publication_type: ProjectFolders::Folder.name)
searched_folders = ProjectFolders::Folder.search_by_all(options[:search])
```
This might include draft folders, or other things you don't want. So may not be the most accurate way to do it. Just thinking of the basic approach.

Reply:
Good point, it doesn't match the `Project` search, though in the docs for pg_search calling the search directly on the model name is how they do it in several places:

```
class Person < ActiveRecord::Base
  include PgSearch::Model
  pg_search_scope :search_by_name, lambda { |name_part, query|
    raise ArgumentError unless [:first, :last].include?(name_part)
    {
      against: name_part,
      query: query
    }
  }
end

person_1 = Person.create!(first_name: "Grant", last_name: "Hill")
person_2 = Person.create!(first_name: "Hugh", last_name: "Grant")

Person.search_by_name :first, "Grant" # => [person_1]
Person.search_by_name :last, "Grant" # => [person_2]
```

Reply:
True. So it probably should work 🤔 
In any case, I think you want to search the folders from the currently filtered collection of admin publications, which should be being passed through the chain of filters.

Reply:
Yeah, that part is right, I updated it to

```
    folders_in_scope = scope.where(publication_type: ProjectFolders::Folder.name)
    searched_folders = folders_in_scope.search_by_all(options[:search])
```

Reply:
I haven't written tests yet but in the FE, the previous version works (shows the right results) and this new version does not. Going to push what I have and maybe we can pair on this, I think I'm hitting the limit of my Ruby. The code seems straightforward enough but would be good to work with someone who does Rails regularly to see how they'd debug and finish the feature (with specs)

Reply:
Sure. Can pair in about an hour from now. :+1: